src/backend/distributed/commands/table.c

```diff
@@ -282,6 +282,15 @@ PostprocessCreateTableStmtForeignKeys(CreateStmt *createStatement)
 	bool missingOk = false;
 	Oid relationId = RangeVarGetRelid(createStatement->relation, NoLock, missingOk);
 
+	if (ShouldCreateTenantSchemaTable(relationId))
+	{
+		/*
+		 * Avoid unnecessarily adding the table into metadata if we will
+		 * distribute it as a tenant table later.
+		 */
+		return;
```

Comment:
still kind of wondering whether we can call ConvertNewTableIfNecessary from here

Reply:
Hmm maybe we can do but would might mean a somewhat big change in the context of this PR --need to understand why we didn't do so for ConvertToManagedTable before.